components/bus/name.go

```diff
@@ -0,0 +1,42 @@
+package bus
+
+import (
+	"fmt"
+	"strings"
+)
+
+// FullyQualifiedStructName name returns object name in format [package].[type name].
+// It ignores if the value is a pointer or not.
+func FullyQualifiedStructName(v any) string {
+	s := fmt.Sprintf("%T", v)
+	s = strings.TrimLeft(s, "*")
+
+	return s
+}
+
+// StructName name returns struct name in format [type name].
+// It ignores if the value is a pointer or not.
+func StructName(v any) string {
+	segments := strings.Split(fmt.Sprintf("%T", v), ".")
+
+	return segments[len(segments)-1]
+}
+
+type namedStruct interface {
+	Name() string
+}
+
+// NamedStruct returns the name from a message implementing the following interface:
+// 		type namedStruct interface {
+// 			Name() string
+// 		}
+// It ignores if the value is a pointer or not.
+func NamedStruct(fallback func(v any) string) func(v any) string {
```

Comment:
I imagine we could panic early in `NamedStruct` if `fallback == nil` (and not in the closure in L40)

Reply:
Or, we could make an opinionated choice like 
```go
if fallback == nil {
    fallback = StructName
}
```